### PR TITLE
Update Jetpack Cloud Backup Time Logic

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Switch `DatePicker` from semi-controlled to controlled component
* Use Moment for `getBackupAttemptsForDate`  
* Add a few tests for `getBackupAttemptsForDate`
* Change `date` state in `BackupPage` to Date

#### Testing instructions

1. run new tests via `npm run test-client -- --testPathPattern jetpack-cloud/sections/backup` and confirm they pass
2. Load up this PR and make sure the component appears for each day that can be selected, and that there are no console errors related to this component
3. Ensure that shuttling back and forth with the date picker changes the date for the backup status component, and that dates with missing backups explain what went wrong in the error message.

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/backups/test/utils.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		const logs = [
+			{
+				activityName: 'rewind__backup_complete_full',
+				activityDate: '2020-03-04T05:02:10.215+00:00',
+			},
+		];
+
+		const filteredLogs = getBackupAttemptsForDate(
+			logs,
+			new Date( '2020-03-03T19:02:10.215+08:00' )
```
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/test/utils.js]:
```suggestion
			new Date( '2020-03-03T19:02:10.215-08:00' )
```

File: client/landing/jetpack-cloud/sections/backups/test/utils.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	test( 'should not filter out rewind complete items on date in other timezone', () => {
+		const logs = [
+			{
+				activityName: 'rewind__backup_complete_full',
+				activityDate: '2020-03-04T05:02:10.215+00:00',
+			},
+		];
+
+		const filteredLogs = getBackupAttemptsForDate(
```
[the remark this answers is not in the document]
Comment (rcoll) [reply client/landing/jetpack-cloud/sections/backups/test/utils.js]: That test is failing for me, either way. Looking into it.

File: client/landing/jetpack-cloud/sections/backups/test/utils.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	test( 'should not filter out rewind complete items on date in other timezone', () => {
+		const logs = [
+			{
+				activityName: 'rewind__backup_complete_full',
+				activityDate: '2020-03-04T05:02:10.215+00:00',
+			},
+		];
+
+		// 2020-03-04T05:02:10 -10:00 = 2020-03-03T29:02:10 and should be considered on the same day
+		const filteredLogs = getBackupAttemptsForDate( logs, '2020-03-03T19:02:10.215-10:00' );
```
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/test/utils.js]: It does work as expected. 2020-03-03T19:02:10.215-**05:00** should pass the test, but it fails.

**Comment by rcoll** [comment]
All works beautiful now. :ship:

File: client/landing/jetpack-cloud/sections/backups/test/utils.js (hunk 3 above)
Comment (cleacos) [inline client/landing/jetpack-cloud/sections/backups/test/utils.js]: Looking for the edge case to test:

```suggestion
		// 2020-03-04T05:02:10 -06:00 = 2020-03-03T23:02:10 and should be considered on the same day
		const filteredLogs = getBackupAttemptsForDate( logs, '2020-03-03T19:02:10.215-06:00' );
```

**Pull request merged**